tools/src/compileProtos.ts

```diff
@@ -344,14 +345,16 @@ export async function generateRootName(directories: string[]): Promise<string> {
   // It's OK to play some guessing game here: if we locate `package.json`
   // with a package name, we'll use it; otherwise, we'll fallback to 'default'.
   for (const directory of directories) {
-    const packageJson = path.resolve(directory, '..', 'package.json');
-    if (fs.existsSync(packageJson)) {
-      const json = JSON.parse((await readFile(packageJson)).toString()) as {
-        name: string;
-      };
-      const name = json.name.replace(/[^\w\d]/g, '_');
-      const hopefullyUniqueName = `${name}_protos`;
-      return hopefullyUniqueName;
+    for (const p of walkUp(path.resolve(directory, '..'))) {
+      const packageJson = path.join(p, 'package.json');
+      if (fs.existsSync(packageJson)) {
+        const json = JSON.parse((await readFile(packageJson)).toString()) as {
+          name: string;
+        };
+        const name = json.name.replace(/[^\w\d]/g, '_');
+        const hopefullyUniqueName = `${name}_protos`;
+        return hopefullyUniqueName;
+      }
     }
   }
   return 'default';
```

Comment:
[unrelated to this change] Wouldn't it be better to use a random name instead of `default` here?

Reply:
Maybe! I didn't write this code though, and perhaps better to do it in a F/U PR

Reply:
I prepared the change in https://github.com/orgads/gax-nodejs/tree/random-id. Since it depends on this commit, I'll open a PR after this one is merged.